pkg/k8shandler/kibana/route.go

```diff
@@ -94,56 +98,39 @@ func (clusterRequest *KibanaRequest) createOrUpdateKibanaRoute() error {
 		return fmt.Errorf("Failure creating Kibana route for %q: %v", cluster.Name, err)
 	}
 
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleLinks() error {
+	cluster := clusterRequest.cluster
+
 	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
 	if err != nil {
 		return err
 	}
 
-	sharedConfig := createSharedConfig(cluster.Namespace, kibanaURL, kibanaURL)
-	utils.AddOwnerRefToObject(sharedConfig, utils.AsOwner(cluster))
-
-	err = clusterRequest.Create(sharedConfig)
-	if err != nil && !errors.IsAlreadyExists(err) {
-		return fmt.Errorf("Failure creating Kibana route shared config: %v", err)
+	if err = clusterRequest.RemoveConsoleLink(kibanaAppConsoleLinkName); err != nil {
```

Comment:
why are we doing a Remove and then Create instead of trying to create and then updating if it already exists?